### PR TITLE
[SPIR-V] correct code for upstreaming

### PR DESCRIPTION
**Pull request by iliya-diyachkov**
The patch contains changes in the code for the latest upstreaming. No difference in testing is expected.

### CODE REVIEW

File: llvm/lib/Target/SPIRV/SPIRVUtils.cpp [hunk 1]
```diff
@@ -298,7 +287,7 @@ static bool isNonMangledOCLBuiltin(StringRef Name) {
          Name == "__translate_sampler_initializer";
 }
 
-std::string isOclOrSpirvBuiltin(StringRef Name) {
+std::string mayBeOclOrSpirvBuiltin(StringRef Name) {
```
Comment (zuban32) [inline llvm/lib/Target/SPIRV/SPIRVUtils.cpp]: The name is misleading, I'd expect this function to return `bool` based on its name. And I don't like 'mayBe' as well, I suppose the result should be definite, or pls add a comment clarifying what's this about

File: llvm/lib/Target/SPIRV/SPIRVTargetMachine.cpp [hunk 2]
```diff
@@ -170,24 +169,22 @@ void SPIRVPassConfig::addPreLegalizeMachineIR() {
   addPass(createSPIRVPreLegalizerPass());
 }
 
-// Use a default legalizer
+// Use a default legalizer.
```
Comment (zuban32) [inline llvm/lib/Target/SPIRV/SPIRVTargetMachine.cpp]: s/a/the

File: llvm/lib/Target/SPIRV/SPIRVTargetMachine.cpp [hunk 3]
```diff
@@ -170,24 +169,22 @@ void SPIRVPassConfig::addPreLegalizeMachineIR() {
...
 bool SPIRVPassConfig::addLegalizeMachineIR() {
   addPass(new Legalizer());
   return false;
 }
 
-// Do not add a RegBankSelect pass, as we only ever need virtual registers
+// Do not add a RegBankSelect pass, as we only ever need virtual registers.
```
Comment (zuban32) [inline llvm/lib/Target/SPIRV/SPIRVTargetMachine.cpp]: s/a/the

File: llvm/lib/Target/SPIRV/SPIRVBuiltins.h [hunk 4]
```diff
@@ -27,13 +28,14 @@ namespace llvm {
...
-/// Register(0) otherwise. 
-/// \p OrigRetTy is the type of the \p OrigRet. 
+/// Register(0) otherwise.
+/// \p OrigRetTy is the type of the \p OrigRet.
 /// \p Args are the arguments of the lowered builtin call.
-std::pair<bool, bool> lowerBuiltin(
-    const StringRef DemangledCall, SPIRV::InstructionSet::InstructionSet Set,
-    MachineIRBuilder &MIRBuilder, const Register OrigRet, const Type *OrigRetTy,
-    const SmallVectorImpl<Register> &Args, SPIRVGlobalRegistry *GR);
+std::pair<bool, bool>
```
Comment (zuban32) [inline llvm/lib/Target/SPIRV/SPIRVBuiltins.h]: Getting back to my original concerns about this, I though that `std::optional<bool>` should be a perfect replacement for this pair. As LLVM currently support C++17 by default, we're ok from that side too.

File: llvm/lib/Target/SPIRV/SPIRVUtils.cpp (hunk 1 above)
Comment (iliya-diyachkov) [reply llvm/lib/Target/SPIRV/SPIRVUtils.cpp]: Ok, probably a more appropriate name is `getOclOrSpirvBuiltinDemangledName()`. The comment in SPIRVUtils.h should be also corrected.

File: llvm/lib/Target/SPIRV/SPIRVBuiltins.h (hunk 4 above)
Comment (zuban32) [reply llvm/lib/Target/SPIRV/SPIRVBuiltins.h]: BTW I became curious to try that myself, so https://reviews.llvm.org/D132802

Comment (iliya-diyachkov) [reply llvm/lib/Target/SPIRV/SPIRVBuiltins.h]: @zuban32, isn't it better to use `llvm::Optional` (llvm/ADT/Optional.h)? It's widely used in llvm. On other hand, `std::optional` is not used at all in llvm/lib (as I see in llvm15 codebase).

**Pull request merged**